ofrak_core/ofrak/gui/script_builder.py

```diff
@@ -0,0 +1,405 @@
+
+        :return: List of strings where each entry is a line in the script
+        """
+        root_resource = await self._get_root_resource(resource)
+        return self._get_script(root_resource.get_id())
+
+    async def commit_to_script(self, resource: Resource):
+        root_resource = await self._get_root_resource(resource)
+        session = self._get_session(root_resource.get_id())
+        for id, name in session.resource_variable_names_queue.items():
+            session.resource_variable_names[id] = name
+        session.actions += session.actions_queue
+        session.actions_queue = []
+        session.resource_variable_names_queue = {}
+
+    async def clear_script_queue(self, resource: Resource):
+        root_resource = await self._get_root_resource(resource)
+        session = self._get_session(root_resource.get_id())
+        session.actions_queue = []
+        session.resource_variable_names_queue = {}
+
+    async def _add_variable(self, resource: Resource) -> str:
+        """
+        Replaces references to a particular resource selected in the GUI with a generated variable
+        name based on uniquely identifying characteristics of the resource. This overcomes the issue
+        of referencing the same resource across OFRAK contexts due to the randomly generated
+        resource IDs changing.
+
+        :param resource: Resource that needs to be uniquely identified in the script
+
+        :return: a unique variable name
+        """
+        if await self._var_exists(resource):
+            return await self._get_variable_from_session(resource)
+
+        root_resource = await self._get_root_resource(resource)
+        if resource.get_id() == root_resource.get_id():
+            await self._add_variable_to_session_queue(resource, "root_resource")
+            return "root_resource"
+
+        parent = await resource.get_parent()
+        if not await self._var_exists(parent):
+            await self._add_variable(parent)
+
+        name = ""
+        # Cannot propagate exceptions to the server as this would interfere with user actions
+        # regardless of whether they're interested in the script. Currently only _get_selector()
+        # and _generate_name() can lead to exceptions raised within ScriptBuilder.
+        try:
+            selector = await self._get_selector(resource)
+            name = await self._generate_name(resource)
+            await self._add_action_to_session_queue(
+                resource,
+                rf"""
+        {name} = {selector}""",
+                ActionType.UNDEF,
+            )
+            await self._add_variable_to_session_queue(resource, name)
+        except SelectableAttributesError as e:
+            name = await self._generate_missing_name(resource, e)
+            LOGGER.exception("Could not find selectable attributes for resource")
+            return name
+        except:
+            LOGGER.exception("Exception raised in add_variable")
```

Comment:
This should probably print out the exception that gets raised.

Reply:
`exception` prints out the message argument as well as the stack trace of the exception that raised it, including the error message of the handled exception. For example:
<img width="544" alt="image" src="https://user-images.githubusercontent.com/24374956/229807450-dcab00d7-fc0f-431b-911d-469604946bc2.png">

Please let me know if this still needs to be changed.